src/main/webapp/app/file-upload-submission/file-upload-submission.component.ts

```diff
@@ -0,0 +1,207 @@
+                        this.submittedFileName = filePath[filePath.length - 1];
+                    }
+                }
+
+                this.isActive =
+                    this.fileUploadExercise.dueDate === undefined || this.fileUploadExercise.dueDate === null || new Date() <= moment(this.fileUploadExercise.dueDate).toDate();
+            },
+            (error: HttpErrorResponse) => this.onError(error),
+        );
+    }
+
+    /**
+     * Uploads a submission file and submits File Upload Exercise
+     */
+    submit() {
+        const file = this.submissionFile;
+
+        if (!this.submission || !file) {
+            return;
+        }
+
+        this.erroredFile = null;
+        this.fileUploaderService.uploadFile(file, file['name'], { keepFileName: true, isSubmission: true }).then(
+            result => {
+                this.submission!.filePath = result.path;
+                this.submitExercise();
+            },
+            error => {
+                console.error('Error during file upload in uploadBackground()', error.message);
+                this.erroredFile = file;
+                this.fileInput.nativeElement.value = '';
+                this.submission!.filePath = null;
+                this.submissionFile = null;
+            },
+        );
+    }
+
+    /**
+     * Submits File Upload Exercise
+     */
+    submitExercise() {
+        const confirmSubmit = window.confirm(this.submissionConfirmationText);
+
+        if (confirmSubmit) {
+            this.submission!.submitted = true;
+            this.fileUploadSubmissionService.update(this.submission!, this.fileUploadExercise.id).subscribe(
+                response => {
+                    this.submission = response.body!;
+                    this.result = this.submission.result;
+
+                    if (this.isActive) {
+                        this.jhiAlertService.success('artemisApp.fileUploadExercise.submitSuccessful');
+                    } else {
+                        this.jhiAlertService.warning('artemisApp.fileUploadExercise.submitDeadlineMissed');
+                    }
+                },
+                err => {
+                    this.jhiAlertService.error('artemisApp.fileUploadExercise.error');
+                    this.submission!.submitted = false;
+                },
+            );
+        }
+    }
+
+    /**
+     * Sets file submission for exercise
+     * @param $event {object} Event object which contains the uploaded file
+     */
+    setFileSubmissionForExercise($event: any): void {
+        if ($event.target.files.length) {
+            this.erroredFile = null;
+            const fileList: FileList = $event.target.files;
+            const submissionFile = fileList[0];
+            const allowedFileExtensions = this.fileUploadExercise.filePattern.split(',');
+            if (allowedFileExtensions.some(extension => submissionFile.name.endsWith(extension))) {
```

Comment:
```suggestion
            if (allowedFileExtensions.some(extension => submissionFile.name.toLowerCase().endsWith(extension))) {
```

Reply:
Without this the submission of test.PDF is not allowed.

Reply:
Did you create a new File Upload Exercise? Because now I lowercase the file pattern on file upload exercise creation / update

Reply:
Yes, you do it on the creation of the FileUploadExercise, but if the file ending of the uploaded file is PDF, the check will be `'PDF'.endsWith('pdf');`, correct?

Reply:
This issue is not resolved. Files with capitalized file endings can still not be uploaded:
![image](https://user-images.githubusercontent.com/28230611/64941338-a0f22b80-d866-11e9-8a42-c4717f4488e4.png)